csrc/python_frontend/fusion_cache.cpp

```diff
@@ -436,12 +437,14 @@ TrieNode* FusionCache::createChild(TrieNode* node, RecordFunctor* rec) {
   } else {
     size_t fusion_id = 0;
     if (rec->recordType() == serde::RecordType::End) {
-      NVF_CHECK(
-          (fusions_.size() + 1) <= max_fusions_,
-          "The number of fusions in nvfuser has exceeded ",
-          max_fusions_,
-          "fusions.  The max_fusions for the FusionCache might need to be ",
-          "increased if the max number is not being exceeded due to an error.");
+      // If the cache is full, free up 1/64th of the entries before we add in
+      // this node.
+      if (fusions_.size() + 1 > max_fusions_) {
+        const long tentative_free = static_cast<long>(max_fusions_) / 64;
+        const long sz_l = static_cast<long>(fusions_.size());
+        const long n_free_nodes = std::clamp(tentative_free, 1L, sz_l);
+        fusions_.erase(fusions_.begin(), fusions_.begin() + n_free_nodes);
```

Comment:
Thanks. This is likely the key insight I am missing that's causing it to break CI ;-)

Reply:
Ryan and I chatted and I'm going to file an issue for larger refactoring. I changed this PR to just increase the size of the cache so that we can unblock dev.